Convert fx.rst to fx.md

Rachel0619 opened the pull request:
Fixes #155023

@pytorchbot label "topic: docs"
@pytorchbot label "topic: not user facing"
@pytorchbot label docathon-h1-2025
@pytorchbot label module: docs

cc @svekars @sekyondaMeta @AlannaBurke

svekars commented:
We need a git mv of fx.rst => fx.md. Can you please make sure the fx.rst is part of this PR.

svekars committed:
Convert fx.rst to fx.md

svekars commented:
can do something like this: https://github.com/pytorch/pytorch/pull/155492 - feel free to grab that PR - this avoids moving code blocks and therefore the number of changes is smaller. Then, plan to submit a follow up PR to move properly fix indentation in those code blocks.

svekars commented on docs/source/fx.md:
```diff
@@ -678,81 +670,78 @@ Going off of the example above, consider the following code:
+want to see the Node’s `input_nodes` and `users`.)
 
-.. _Available Debuggers:
+(Available Debuggers)=
```
```suggestion
(available_debuggers)=
```

svekars commented on docs/source/fx.md:
```diff
@@ -572,38 +566,36 @@ your code and examine it from there.
 
-Debugging the Transformation
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#### Debugging the Transformation
```
```suggestion
### Debugging the Transformation
```

svekars commented on docs/source/fx.md:
```diff
@@ -207,212 +211,208 @@ can be found below.
 
-Introduction
-^^^^^^^^^^^^^^^^
+#### Introduction
```
```suggestion
### Introduction
```

svekars commented on docs/source/fx.md:
```diff
@@ -421,33 +421,31 @@ disprove correctness. Then, inspect and debug the generated code. Then, debug th
 
-Common Pitfalls in Transform Authoring
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#### Common Pitfalls in Transform Authoring
```
```suggestion
### Common Pitfalls in Transform Authoring
```

svekars commented on docs/source/fx.md:
```diff
@@ -421,33 +421,31 @@ disprove correctness. Then, inspect and debug the generated code. Then, debug th
 
-Checking Correctness of Modules
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#### Checking Correctness of Modules
```
```suggestion
### Checking Correctness of Modules
```

svekars commented on docs/source/fx.md:
```diff
@@ -474,38 +472,35 @@ example:
 
-Debugging the Generated Code
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#### Debugging the Generated Code
```
```suggestion
### Debugging the Generated Code
```

svekars commented on docs/source/fx.md:
```diff
@@ -474,38 +472,35 @@ example:
-Invoke ``pdb`` to step into the running program. Although the code that
-represents the :class:`Graph` is not in any source file, we can still step
-into it manually using ``pdb`` when the forward pass is invoked.
+## Use `pdb`
```
```suggestion
#### Use `pdb`
```

svekars commented on docs/source/fx.md:
```diff
@@ -530,17 +525,16 @@ into it manually using ``pdb`` when the forward pass is invoked.
-
-Print the Generated Code
-~~~~~~~~~~~~~~~~~~~~~~~~~~~
+## Print the Generated Code
```
```suggestion
#### Print the Generated Code
```

svekars commented on docs/source/fx.md:
````diff
@@ -572,38 +566,36 @@ your code and examine it from there.
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-:meth:`GraphModule.to_folder` is a method in ``GraphModule`` that allows
+```
+## Use the `to_folder` Function From `GraphModule`
````
```suggestion
#### Use the `to_folder` Function From `GraphModule`
```

svekars commented on docs/source/fx.md:
```diff
@@ -678,81 +670,78 @@ Going off of the example above, consider the following code:
 
-Limitations of Symbolic Tracing
--------------------------------
+### Limitations of Symbolic Tracing
```
```suggestion
## Limitations of Symbolic Tracing
```

svekars commented on docs/source/fx.md:
```diff
@@ -678,81 +670,78 @@ Going off of the example above, consider the following code:
 
-Dynamic Control Flow
-^^^^^^^^^^^^^^^^^^^^
+#### Dynamic Control Flow
```
```suggestion
### Dynamic Control Flow
```

svekars commented on docs/source/fx.md:
```diff
@@ -766,23 +755,22 @@ For example, let’s examine the following program:
 
-Static Control Flow
-~~~~~~~~~~~~~~~~~~~
+## Static Control Flow
```
```suggestion
### Static Control Flow
```

svekars commented on docs/source/fx.md:
```diff
@@ -474,38 +472,35 @@ example:
     """
```
ah, missed the closing backticks here for the code block

Rachel0619 committed:
Update docs/source/fx.md

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

Rachel0619 committed:
Update docs/source/fx.md

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

Rachel0619 committed:
Update docs/source/fx.md

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

Rachel0619 committed:
Update docs/source/fx.md

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

Rachel0619 committed:
Update docs/source/fx.md

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

Rachel0619 committed:
Update docs/source/fx.md

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

Rachel0619 committed:
Update docs/source/fx.md

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

Rachel0619 committed:
Update docs/source/fx.md

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

Rachel0619 committed:
Update docs/source/fx.md

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

Rachel0619 committed:
Update docs/source/fx.md

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

Rachel0619 committed:
Update docs/source/fx.md

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

Rachel0619 committed:
Update docs/source/fx.md

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

Rachel0619 committed:
Update docs/source/fx.md

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

Rachel0619 committed:
Update docs/source/fx.md

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

Rachel0619 committed:
Update docs/source/fx.md

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

Rachel0619 committed:
Update docs/source/fx.md

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

Rachel0619 committed:
Update docs/source/fx.md

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

Rachel0619 committed:
fix syntax errors in fx.md

Rachel0619 commented:
Just fixed some syntax errors that caused warnings.

- changed `{eval_rst}` to `{eval-rst}`
- added `__init__` to `automethod`
- adding missing closing backticks for the code block

svekars committed:
Minort formatting fixes

svekars commented on docs/source/fx.md:
```diff
@@ -421,33 +412,31 @@ disprove correctness. Then, inspect and debug the generated code. Then, debug th
   for example, can cause unexpected nondeterminism. An example is iterating
-  over a set of ``Node``\ s to insert them into a ``Graph``. Because the
-  ``set`` data type is unordered, the ordering of the operations in the output
+  over a set of `Node`\ s to insert them into a `Graph`. Because the
```
```suggestion
  over a set of `Node` s to insert them into a `Graph`. Because the
```

svekars committed:
Update docs/source/fx.md

svekars committed:
Update docs/source/fx.md

svekars reviewed (approved):
Ok, this looks good, after this PR is merged please submit a follow up to fix indentation in code blocks.

Rachel0619 commented:
Sure, I'll work on that later. Thank you so much! @svekars

svekars commented:
@pytorchbot merge -f "Docs changes only. Doc signal is green."
